### PR TITLE
Reminders

### PR DESCRIPTION
**Pull request by superstarcoder**

### CODE REVIEW

File: src/commands/tasks/listTasks.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+import {getLists, displayTasks, List} from '../../tasks/trelloTasks'
+import { client } from '../../index';
+
+
+@ApplyOptions<CommandOptions>({
+	description: 'lists tasks of a user',
+	aliases: ['ls']
+})
+export class ListTasks extends Command {
+	public async messageRun(message: Message, _args: Args) {
```
Comment (nmathira) [inline src/commands/tasks/listTasks.ts]: remove unused _args and unimport it cuz you're not using it

File: src/index.ts [hunk 2]
```diff
@@ -26,12 +27,37 @@ const main = async () => {
...
+ */
+async function sendReminder(hour:number, minute: number, second: number) {
+  while (true) {
+    let now:any = new Date();
+    let then:any = new Date(now.getFullYear(), now.getMonth(), now.getDate(), hour, minute, second, 0)
+    let millisTill:number = then - now;
+    if (millisTill < 0) {
+        millisTill += 1000*60*60*24; // it's after x am/pm, try x am/pm tomorrow.
+    }
+    console.log("set timer for "+millisTill+"ms")
```
Comment (nmathira) [inline src/index.ts]: use container.logger to log instead of console

File: src/index.ts [hunk 3]
```diff
@@ -26,12 +27,37 @@ const main = async () => {
...
+
+/**
+ * sends reminders at a certain time everyday in a specific channel
+ * @param hour 
+ * @param minute 
+ * @param second 
+ */
+async function sendReminder(hour:number, minute: number, second: number) {
+  while (true) {
+    let now:any = new Date();
```
Comment (nmathira) [inline src/index.ts]: let now: Date = new Date(); I think?

File: .gitignore [hunk 4]
```diff
@@ -24,6 +24,7 @@ config.ts
 yarn.lock
 
 # Environment variables
+.env
```
Comment (nmathira) [inline .gitignore]: use src/.env, or use .env only, I don't want both env files

File: src/index.ts (hunk 3 above)
Comment (nmathira) [reply src/index.ts]: you should also try and use consts where you can, cuz safety 😍

File: src/index.ts [hunk 5]
```diff
@@ -26,12 +27,37 @@ const main = async () => {
...
+    let then:any = new Date(now.getFullYear(), now.getMonth(), now.getDate(), hour, minute, second, 0)
+    let millisTill:number = then - now;
```
Comment (nmathira) [inline src/index.ts]: inferenced type

File: src/index.ts (hunk 3 above)
Comment (nmathira) [reply src/index.ts]: wait this is an inferenced type isn't it

File: src/tasks/trelloTasks.ts [hunk 6]
```diff
@@ -0,0 +1,145 @@
+import type { SapphireClient } from "@sapphire/framework";
+import { ColorResolvable, MessageEmbed } from "discord.js";
+import dotenv from "dotenv";
```
Comment (nmathira) [inline src/tasks/trelloTasks.ts]: use one .env manager pls

File: src/tasks/trelloTasks.ts [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+
+const fetch = require('node-fetch');
+const membersJson = require("../../json/members.json")
+const listsToIgnoreJson = require("../../json/listsToIgnore.json")
+
+dotenv.config()
+
+// super useful types that we'll be using
+export type Member = {trelloId: string, discordId:string, name:string};
```
Comment (nmathira) [inline src/tasks/trelloTasks.ts:10]: can we make a extra file somewhere for types? I find that easier to look through instead. something like src/typings/index.ts, and you put all of your types in there

File: src/tasks/trelloTasks.ts [hunk 8]
```diff
@@ -0,0 +1,145 @@
...
+export type Card = {name: string, daysDueIn: number, members: Member[], dueString: string, desc: string}
+export type List = {name:string, trelloId: string, cards: Card[]}
+
+
+// helper function to make rounding easier
+function round(value:number, precision:number) {
+  var multiplier = Math.pow(10, precision || 0);
```
Comment (nmathira) [inline src/tasks/trelloTasks.ts:17]: change var for let/const

File: src/tasks/trelloTasks.ts [hunk 9]
```diff
@@ -0,0 +1,145 @@
...
+  return Math.round(value * multiplier) / multiplier;
+}
+
+
+// takes a list of trello ids of members and uses members.json to return a list of Members
+function getMembers(trelloIdMembers: string[]) {
```
Comment (nmathira) [inline src/tasks/trelloTasks.ts:23]: return type?

File: src/tasks/trelloTasks.ts [hunk 10]
```diff
@@ -0,0 +1,145 @@
...
+
+/**
+ * takes in list id and returns a list structure
+ * @param listId 
+ * @returns 
+ */
+async function getList(listId: string){
+
+  let url = `https://api.trello.com/1/lists/${listId}/cards?key=${process.env.TRELLO_KEY}&token=${process.env.TRELLO_TOKEN}`
+  console.log("opening url "+url)
```
Comment (nmathira) [inline src/tasks/trelloTasks.ts]: container.logger

File: src/tasks/trelloTasks.ts [hunk 11]
```diff
@@ -0,0 +1,145 @@
...
+          await getList(element.id).then(
+          myList => { 
+            lists.push(myList)
+            // console.log(JSON.stringify(myList,null,'  '))
+          }
+        )
+      }
+    }
+  })
+  console.log(JSON.stringify(lists,null,'  '))
```
Comment (nmathira) [inline src/tasks/trelloTasks.ts]: container.logger

File: src/tasks/trelloTasks.ts [hunk 12]
```diff
@@ -0,0 +1,145 @@
...
+
+
+/**
+ * takes in board id and returns board/lists (list of List objects)
+ * @param boardId 
+ * @returns 
+ */
+export async function getLists(boardId:string){
+  let url: string = `https://api.trello.com/1/boards/${boardId}/lists/all?key=${process.env.TRELLO_KEY}&token=${process.env.TRELLO_TOKEN}`
+  console.log("opening url "+url)
```
Comment (nmathira) [inline src/tasks/trelloTasks.ts]: make this a container.logger.trace so that it doesn't get printed evry time

**Review by nmathira — CHANGES_REQUESTED** [review]
quick changes, nothing major

File: src/index.ts (hunk 5 above)
Comment (superstarcoder) [reply src/index.ts]: nope it returns a number (ms between the two times)

File: src/tasks/trelloTasks.ts (hunk 7 above)
Comment (superstarcoder) [reply src/tasks/trelloTasks.ts:10]: kk ill do that in a diff pr tho

**Review by nmathira — APPROVED** [review]

**Comment by nmathira** [comment]
@KaustubhKhulbe approved

**Pull request merged**